frontend/server/src/DAO/QualityNominations.php

```diff
@@ -123,6 +123,73 @@ public static function reviewerHasQualityTagNominatedProblem(
         ) > 0;
     }
 
+    /**
+     * Returns the contents and id of a nomination for a given problem and user.
+     */
```

Comment:
Aquí, a pesar de que ya le estés indicando a php que tu función va a regresar un array, psalm requiere saber de qué esta compuesto dicho array. Así que debería ir algo así:
```suggestion
    /**
     * Returns the contents and id of a nomination for a given problem and user.
     * @return array{contents: array{quality_seal: bool, tag: string}, qualitynomination_id: int}|null
     */
```
como aquí estás indicando que puede regresar el array con esos elementos o `null`, entonces tendrías que cambiar L157 por:
```php
            return null;
```